pennylane/ops/op_math/adjoint.py

```diff
@@ -221,6 +221,8 @@ def __new__(cls, base=None, do_queue=None, id=None):
         True
         >>> type(Adjoint(op)) == type(Adjoint(op))
         True
+        >>> isinstance(Adjoint(op), type(Adjoint(op)))
+        True
```

Comment:
I feel like this is tautological, since you get they type of `Adjoint(op)` from `type(Adjoint(op))`. Might it make more sense to say `isinstance(Adjoint(op), Adjoint)`? 

Even if the type somehow weren't 'Adjoint', this would return True, right?

Reply:
Indeed if this was a typical class, it would be redundant. However, in this case, `Adjoint` is a metaclass that returns a `type("Adjoint", base_classes, dict)` such that if we naively created two adjoint operators one after the other, these would have two different types despite them both being `Adjoint`. This would cause `isinstance(Adjoint(op), type(Adjoint(op)))` to be `False`. For this reason, we store the first of each different type in hidden variables to reuse them for subsequent instances. This ensures that `isinstance(Adjoint(op), type(Adjoint(op)))` is `True`. I spent some time to figure this out, which is why I added the line 😅 

That said, I could actually add `isinstance(Adjoint(op), Adjoint)` too as an additional example.